src/Service/Configurations/RuntimeConfigValidator.cs

```diff
@@ -239,25 +239,70 @@ public static void ValidateEntitiesDoNotGenerateDuplicateQueriesOrMutation(IDict
         /// have GraphQL configuration: when entity.GraphQL == false or null.
         /// </summary>
         /// <seealso cref="https://spec.graphql.org/October2021/#Name"/>
-        /// <param name="runtimeConfig"></param>
-        public static void ValidateEntityNamesInConfig(Dictionary<string, Entity> entityCollection)
+        /// <param name="runtimeConfig">The runtime configuration.</param>
+        public static void ValidateEntityConfiguration(RuntimeConfig runtimeConfig)
         {
-            foreach (string entityName in entityCollection.Keys)
+            // Stores the unique rest paths configured for different entities present in the config.
+            HashSet<string> restPathsForEntities = new();
+
+            foreach (KeyValuePair<string, Entity> Entity in runtimeConfig.Entities)
             {
-                Entity entity = entityCollection[entityName];
+                string entityName = Entity.Key;
+                Entity entity = Entity.Value;
+
+                // If no custom rest path is defined for the entity, we default it to the entityName.
+                string pathForEntity = entityName;
+                if (entity.Rest is not null)
+                {
+                    JsonElement restJsonElement = JsonSerializer.SerializeToElement(entity.Rest);
+
+                    // We do the validation for rest path only if the 'rest' property maps to a json object.
+                    if (restJsonElement.ValueKind is JsonValueKind.Object)
+                    {
+                        // Since 'path' is an optional property, we skip validation if its absent.
+                        if (restJsonElement.TryGetProperty(RestEntitySettings.PROPERTY_PATH, out JsonElement pathElement))
+                        {
+                            pathForEntity = ValidateAndGetRestPathForEntity(entityName, pathElement);
+                        }
+
+                        // Since 'methods' is an optional property, we skip validation if its absent.
+                        if (restJsonElement.TryGetProperty(RestStoredProcedureEntitySettings.PROPERTY_METHODS, out JsonElement methodsElement))
+                        {
+                            ValidateRestMethodsForEntity(entityName, methodsElement, entity);
+                        }
+                    }
+                    else if (restJsonElement.ValueKind is not JsonValueKind.True && restJsonElement.ValueKind is not JsonValueKind.False)
+                    {
+                        throw new DataApiBuilderException(
+                        message: $"The 'rest' property for entity: {entityName} can only be a boolean value or a json object.",
+                        statusCode: HttpStatusCode.ServiceUnavailable,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError
+                        );
+                    }
+                }
+
+                if (!restPathsForEntities.Add(pathForEntity))
```

Comment:
What if for EntityA rest is set to false, and for EntityB the path is set to EntityA, will it allow such a config or will it fail? What should be the desired behavior?